fieldservice_maintenance/models/fsm_equipment.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (C) 2018 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class FSMEquipment(models.Model):
+    _inherit = "fsm.equipment"
+
+    maintenance_equipment_id = fields.Many2one(
+        "maintenance.equipment",
+        string="Related Maintenance Equipment",
+        readonly=True,
+        ondelete="restrict",
+        delegate=True,
+        auto_join=True,
+        index=True,
+    )
+
+    @api.model
+    def create(self, vals):
+        maintenance_equipment_id = self.env["maintenance.equipment"].create(
+            self._prepare_maintenance_vals(vals)
+        )
+        if maintenance_equipment_id:
```

Comment:
The "if" here is not necessary because if the create doesn't work, you'll have an exception